File: 13-List-Comprehensions/21-More-List-Comprehension.py
# part 1
list1 = [1,2,3,4]
list2 = [3,4,5,6]

answer = [item for item in list1 if item in list2]

# part 2
teachers = ["Elie", "Tim", "Matt"]

answer2 = [teacher[::-1].lower() for teacher in teachers]

print(answer2)

File: 13-List-Comprehensions/21-More-List-Comprehension.js
let list1 = [1,2,3,4]
let list2 = [3,4,5,6]

// ===== part 1 =====

// using vanilla
let answer = list1.filter(item => list2.includes(item))


// using lodash
const intersection = require('lodash/intersection')

let answer = intersection(list1, list2)


// ===== part 2 =====
let teachers = ["Elie", "Tim", "Matt"]

function reverse(str) {
    return str.split('').reverse().join('')
}

let answer2 = teachers.map(teacher => reverse(teacher).toLowerCase())
